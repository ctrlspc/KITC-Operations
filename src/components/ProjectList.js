import React from 'react';
import ProjectListItem from './ProjectListItem';

export default (props) => (
  <div>
    <div className="list-header">
      <div className="show-for-mobile">Projects</div>
      <div className="show-for-desktop">Project</div>
      <div className="show-for-desktop">Type</div>
    </div>
    <div className="list-body">
      {
        !props.projects || props.projects.length === 0 ? (
          <p>No Projects!</p>
        ) : (
          props.projects.map((project) => <ProjectListItem key={project.id} {...project}/>)
        )
      }
    </div>
  </div>
)