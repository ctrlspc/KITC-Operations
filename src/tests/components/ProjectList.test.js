import React from 'react';
import { shallow } from 'enzyme';
import ProjectList from '../../components/ProjectList';

test('should render the ProjectList component correctly', () => {
  const wrapper = shallow (<ProjectList />);
  expect(wrapper).toMatchSnapshot();
});