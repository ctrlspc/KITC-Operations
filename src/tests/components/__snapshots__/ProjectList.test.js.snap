// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the ProjectList component correctly 1`] = `
<div>
  <div
    className="list-header"
  >
    <div
      className="show-for-mobile"
    >
      Projects
    </div>
    <div
      className="show-for-desktop"
    >
      Project
    </div>
    <div
      className="show-for-desktop"
    >
      Type
    </div>
  </div>
  <div
    className="list-body"
  >
    <p>
      No Projects!
    </p>
  </div>
</div>
`;
